tests/bson/bug1598-002.phpt

```diff
@@ -1,5 +1,8 @@
 --TEST--
 PHPC-1598: BSON type get_gc should delegate to zend_std_get_properties
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_php_version('>=', '8.1.99'); ?>
```

Comment:
Necessary since "< 8.2.0" would allow the test to run on pre-release versions, correct?

What's the actual reason that we can't run this on 8.2+? Is it due to dynamic creation of the `a` property on each BSON object? I noted from https://wiki.php.net/rfc/deprecate_dynamic_properties that stdClass isn't restricted.

Reply:
> Necessary since "< 8.2.0" would allow the test to run on pre-release versions, correct?

I remember prior art for doing this, but I can also change it to `< 8.2.0`.

> What's the actual reason that we can't run this on 8.2+? Is it due to dynamic creation of the `a` property on each BSON object? I noted from https://wiki.php.net/rfc/deprecate_dynamic_properties that stdClass isn't restricted.

That is correct - starting with PHP 8.2 creating dynamic properties on classes emits a deprecation notice, with the functionality being removed in PHP 9.

Reply:
> I remember prior art for doing this, but I can also change it to < 8.2.0.

Possible misunderstanding here, but wouldn't `skip_if_php_version('>', '8.2.0');` cause test failures since the test would then be allowed to run on 8.2 pre-release versions? I think it's correct as-is, but was just asking for confirmation.